File: 98-validate-binary-search-tree/validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean isBST = true;
    TreeNode prev = null;
    void helper(TreeNode root){
        if(root == null){
            return;
        }
        helper(root.left);
        if (prev != null && root.val <= prev.val){
            isBST = false;
            return;
        }
        prev = root;
        helper(root.right);
    }
    public boolean isValidBST(TreeNode root) {
        helper(root);
        return isBST;
    }
}

File: 98-validate-binary-search-tree/validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        result = True
        prev = None
        def helper(root):
            nonlocal result
            nonlocal prev
            if not root:
                return None
            helper(root.left)
            if prev and prev.val >= root.val:
                result = False
                return
            prev = root
            helper(root.right)
        helper(root)
        return result

        
        

